Array out-of-bounds warning in DataHeader.h

I see the following warning, which probably needs to be fixed or silenced:

In file included from /Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/test/dataHeaderTest.cxx:5:
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:200:11: warning: array index 7 is past the end of the array
      (which contains 7 elements) [-Warray-bounds]
         (str[1+pos] && sizeof(T) >= 4 ? ((T) str[2+pos] << (sizeof(T) >= 4 ? 16 : 0) |

Here is the full log:

In file included from /Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/test/dataHeaderTest.cxx:5:
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:200:11: warning: array index 7 is past the end of the array
      (which contains 7 elements) [-Warray-bounds]
         (str[1+pos] && sizeof(T) >= 4 ? ((T) str[2+pos] << (sizeof(T) >= 4 ? 16 : 0) |
          ^   ~~~~~
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:576:130: note: in instantiation of function template specialization
      'AliceO2::Header::String2<unsigned long long, 7, 6, false>' requested here
  constexpr DataDescription(const char (&desc)[NN]) : itg{String2<uint64_t, NN, 0, true>(desc), (Internal::strLength(desc) > 8 ? String2<uint64_t, NN, std::conditiona...
                                                                                                                                 ^
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/test/dataHeaderTest.cxx:60:23: note: in instantiation of function template specialization
      'AliceO2::Header::DataDescription::DataDescription<7>' requested here
      DataDescription desc("ITSRAW");
                      ^
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:191:21: note: array 'str' declared here
constexpr T String2(const char (&str)[N])
                    ^
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:201:11: warning: array index 8 is past the end of the array
      (which contains 7 elements) [-Warray-bounds]
         (str[2+pos] && sizeof(T) >= 4 ? ((T) str[3+pos] << (sizeof(T) >= 4 ? 24 : 0) |
          ^   ~~~~~
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:191:21: note: array 'str' declared here
constexpr T String2(const char (&str)[N])
                    ^
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:202:11: warning: array index 9 is past the end of the array
      (which contains 7 elements) [-Warray-bounds]
         (str[3+pos] && sizeof(T) >= 8 ? ((T) str[4+pos] << (sizeof(T) >= 8 ? 32 : 0) |
          ^   ~~~~~
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:191:21: note: array 'str' declared here
constexpr T String2(const char (&str)[N])
                    ^
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:203:11: warning: array index 10 is past the end of the array
      (which contains 7 elements) [-Warray-bounds]
         (str[4+pos] && sizeof(T) >= 8 ? ((T) str[5+pos] << (sizeof(T) >= 8 ? 40 : 0) |
          ^   ~~~~~
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:191:21: note: array 'str' declared here
constexpr T String2(const char (&str)[N])
                    ^
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:204:11: warning: array index 11 is past the end of the array
      (which contains 7 elements) [-Warray-bounds]
         (str[5+pos] && sizeof(T) >= 8 ? ((T) str[6+pos] << (sizeof(T) >= 8 ? 48 : 0) |
          ^   ~~~~~
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:191:21: note: array 'str' declared here
constexpr T String2(const char (&str)[N])
                    ^
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:205:11: warning: array index 12 is past the end of the array
      (which contains 7 elements) [-Warray-bounds]
         (str[6+pos] && sizeof(T) >= 8 ? ((T) str[7+pos] << (sizeof(T) >= 8 ? 56 : 0) )
          ^   ~~~~~
/Users/swenzel/git/sw/SOURCES/O2/0_O2_DAQ/0/DataFormats/Headers/include/Headers/DataHeader.h:191:21: note: array 'str' declared here
constexpr T String2(const char (&str)[N])
                    ^

--- a/DataFormats/Headers/include/Headers/DataHeader.h
+++ b/DataFormats/Headers/include/Headers/DataHeader.h
@@ -573,7 +573,7 @@ struct DataDescription {
   // note: no operator=(const char*) as this potentially runs into trouble with this
   // general pointer type, use: somedesc = DataDescription("SOMEDESCRIPTION")
   template<std::size_t NN>
-  constexpr DataDescription(const char (&desc)[NN]) : itg{String2<uint64_t, NN, 0, true>(desc), (Internal::strLength(desc) > 8 ? String2<uint64_t, NN, std::conditional< (NN > 8), Internal::intWrapper<8>, Internal::intWrapper<NN-1>>::type::value >(desc) : 0)} {
+  constexpr DataDescription(const char (&desc)[NN]) : itg{String2<uint64_t, NN, 0, true>(desc), (Internal::strLength(desc) > 8 ? String2<uint64_t, NN, std::conditional< (NN > 8), Internal::intWrapper<8>, Internal::intWrapper<0>>::type::value >(desc) : 0)} {
     // the initializer implements a number of compile time checks
     // - for the conversion of the first part of the string to the first 64bit field
     //   the check for the string length needs to be disabled as it is naturally longer

--- a/DataFormats/Headers/test/dataHeaderTest.cxx
+++ b/DataFormats/Headers/test/dataHeaderTest.cxx
@@ -60,10 +60,20 @@ namespace AliceO2 {
       DataDescription desc("ITSRAW");
       BOOST_CHECK(strcmp(desc.str, "ITSRAW")==0);
 
+      // checking the corresponding integer value
+      // the upper part must be 0 since the string has only up tp 8 chars
+      // lower part corresponds to reverse ITSRAW
+      //                       W A R S T I
+      uint64_t itgDesc = 0x0000574152535449;
+      BOOST_CHECK(desc.itg[0] == itgDesc);
+      BOOST_CHECK(desc.itg[1] == 0);
+
       BOOST_CHECK(desc == "ITSRAW");
 
       DataDescription desc2(test);
       BOOST_CHECK(strcmp(desc2.str, "ITSRAW")==0);
+      // the upper part must be 0 since the string has only up tp 8 chars
+      BOOST_CHECK(desc2.itg[1] == 0);
 
       BOOST_CHECK(desc2 == "ITSRAW");
 

